File: src/main/java/io/github/eone666/telegramjoinnotifications/events/PlayerJoin.java
package io.github.eone666.telegramjoinnotifications.events;

import io.github.eone666.telegramjoinnotifications.events.utils.Telegram;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

public class PlayerJoin implements Listener {

    private Telegram _tg = null;
    public PlayerJoin(Telegram tg){
        _tg = tg;
    }
    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        String playerName = event.getPlayer().getName();
        _tg.SendMessage(String.format("%s has joined the game",playerName ));
    }
}

File: src/main/kotlin/io/github/eone666/telegramnotifier/listeners/PlayerJoin.kt
package io.github.eone666.telegramnotifier.listeners

import io.github.eone666.telegramnotifier.pluginInstance
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class PlayerJoin : Listener {
    @EventHandler
    suspend fun onPlayerJoin(event: PlayerJoinEvent) {
        if(pluginInstance.config.isPluginConfigured.get()){
            pluginInstance.notifications.join(event.player)
            pluginInstance.playersList.add(event.player)
        }
    }
}